opm/common/data/Wells.hpp

```diff
@@ -0,0 +1,88 @@
+/*
+  Copyright 2016 Statoil ASA.
+
+  This file is part of the Open Porous Media project (OPM).
+
+  OPM is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  OPM is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with OPM.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#ifndef COMMON_WELLRESULTS_HPP
+#define COMMON_WELLRESULTS_HPP
+
+#include <map>
+#include <string>
+#include <vector>
+
+namespace Opm {
+
+    namespace data {
+
+    /*
+     * A triple with named fields describing production rates or totals for a
+     * well or perforation. The vector-based constructor is *strictly* for
+     * convenience; simulators often store simulation data in fortran-style
+     * column-oriented vectors. The order of the rates are a shared assumption.
+     */
+    struct production {
+        production() = default;
+        production( double, double, double );
+        production( const std::vector< double >& vec, size_t index,
+                               size_t wat_offset = 0,
+                               size_t oil_offset = 1,
+                               size_t gas_offset = 2 );
+
+        double wat;
```

Comment:
I made it vague on purpose, so that the triple could be used to represent both rate and totals, since this essentially just is a dump of doubles anyway. Now, if we were to actually follow up onto this to its logical conclusion we can introduce distinct *types* for rate and total and make them incompatible.